### PR TITLE
ci: Add manual benchmark step for trace-agent

### PR DESCRIPTION
**Pull request by ajgajg1134**
### What does this PR do?
This PR adds a manual step job in gitlab to run some benchmarks on the APM Benchmarking platform. This can be used to identify regressions (or improvements) on PRs and will auto-comment a simplified version of the results after running.

### Motivation
We want to be able to more easily identify performance changes at the PR level

### Describe how to test/QA your changes
no QA necessary, doesn't change any customer facing things, just a new gitlab job for us

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ahmed-mez — APPROVED** [review]

File: .gitlab/benchmarks/scripts/analyze-results.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
```
Comment (bkabrda) [inline .gitlab/benchmarks/scripts/analyze-results.sh:1]: We already have a `test/benchmarks/` directory which is used for some benchmark-related files. Could you:
* Create a subdirectory there for your files
* Add yourself as CODEOWNERS of this subdirectory
* Move all the scripts there
We'd like to keep the `.gitlab/` directory for Gitlab config files only

File: tasks/trace_agent.py [hunk 2]
```diff
@@ -127,6 +127,7 @@ def cross_compile(ctx, tag=""):
 
 @task
 def benchmarks(ctx, bench, output="./trace-agent.benchmarks.out"):
+
```
Comment (bkabrda) [inline tasks/trace_agent.py:130]: Nit: why do we need to add this extra newline?

**Review by bkabrda — CHANGES_REQUESTED** [review]
Hey :wave: thanks for the PR. I think overall it looks good, but there are couple minor things I'd like to see addressed before I approve - see the comments inline.

File: .gitlab/benchmarks/benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+variables:
+  # This base image is created here: https://gitlab.ddbuild.io/DataDog/apm-reliability/relenv-microbenchmarking-platform
+  BASE_CI_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/relenv-microbenchmarking-platform:trace-agent
+  INDEX_FILE: index.txt
```
[the remark this answers is not in the document]
Comment (ajgajg1134) [reply .gitlab/benchmarks/benchmarks.yml]: good point, I think this was copy-pasted in and is unused. I've removed it

File: .gitlab/benchmarks/benchmarks.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+variables:
+  # This base image is created here: https://gitlab.ddbuild.io/DataDog/apm-reliability/relenv-microbenchmarking-platform
+  BASE_CI_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/relenv-microbenchmarking-platform:trace-agent
```
[the remark this answers is not in the document]
Comment (ajgajg1134) [reply .gitlab/benchmarks/benchmarks.yml]: 👍 yeah, unlikely we'll need it elsewhere (and if we do we can re-add a var)

File: .gitlab/benchmarks/scripts/analyze-results.sh (hunk 1 above)
Comment (ajgajg1134) [reply .gitlab/benchmarks/scripts/analyze-results.sh]: Great idea, I've created a subdir called `apm_scripts` there and assigned us code ownership

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -352,6 +352,7 @@
 /test/kitchen/test/integration/win-user/ @DataDog/windows-agent
 /test/system/                           @DataDog/agent-shared-components
 /test/system/dogstatsd/                 @DataDog/agent-metrics-logs
+/test/benchmarks/apm_scripts/           @DataDog/agent-apm
```
Comment (bkabrda) [inline .github/CODEOWNERS:355]: One last request - could you also add your team as owner of the `.gitlab/benchmars/benchmars.yml` file? I forgot to ask for it during the first review round. I'll approve when it's done. Thanks!

Comment (ajgajg1134) [reply .github/CODEOWNERS:357]: Done! 👍

**Review by bkabrda — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

**Pull request merged**